--- origsrc/krdc-16.08.3/vnc/vncclientthread.cpp	2016-11-06 17:26:22.000000000 -0600
+++ src/krdc-16.08.3/vnc/vncclientthread.cpp	2016-11-14 00:26:10.932314500 -0600
@@ -607,23 +607,29 @@ void VncClientThread::clientSetKeepalive
         return;
     }
 
+#ifdef TCP_KEEPIDLE
     optval = m_keepalive.intervalSeconds;
     if (setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPIDLE, &optval, optlen) < 0) {
         qCritical(KRDC) << "setsockopt(TCP_KEEPIDLE)" << strerror(errno);
         return;
     }
+#endif
 
+#ifdef TCP_KEEPINTVL
     optval = m_keepalive.intervalSeconds;
     if (setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPINTVL, &optval, optlen) < 0) {
         qCritical(KRDC) << "setsockopt(TCP_KEEPINTVL)" << strerror(errno);
         return;
     }
+#endif
 
+#ifdef TCP_KEEPCNT
     optval = m_keepalive.failedProbes;
     if(setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPCNT, &optval, optlen) < 0) {
         qCritical(KRDC) << "setsockopt(TCP_KEEPCNT)" << strerror(errno);
         return;
     }
+#endif
     m_keepalive.set = true;
     qCDebug(KRDC) << "TCP keepalive set";
 }
